zjit/src/hir.rs

```diff
@@ -3112,6 +3121,52 @@ mod opt_tests {
         "#]]);
     }
 
+    #[test]
+    fn test_param_forms_get_bb_param() {
+        eval("
+            def rest(*array) = array
+            def kw(k:) = k
+            def kw_rest(**k) = k
+            def post(*rest, post) = post
+            def block(&b) = nil
+            def forwardable(...) = nil
+        ");
+
+        assert_optimized_method_hir("rest", expect![[r#"
+            fn rest:
+            bb0(v0:ArrayExact):
+              Return v0
+        "#]]);
+        // extra hidden param for number of specified keywords
+        assert_optimized_method_hir("kw", expect![[r#"
+            fn kw:
+            bb0(v0:BasicObject, v1:BasicObject):
```

Comment:
I don't know that this hidden param with the count should be reified into a HIR register if it's not being manipulated by Ruby code

Reply:
oh boy

is this the `kw_bits` thing?

Reply:
Yep. At some point maybe DCE can also drop unused parameters?

Reply:
Perhaps!